Usage of wallet's inherited EventEmitter .on does not compile

This error is shown

```
src/channel-manager.ts:436:19 - error TS2349: This expression is not callable.
  Each member of the union type '(<T extends "channelUpdated">(event: T, fn: (...args: ArgumentMap<EventEmitterType>[Extract<T, "channelUpdated">]) => void, context?: any) => SingleThreadedWallet) | (<T extends "channelUpdated">(event: T, fn: (...args: ArgumentMap<...>[Extract<...>]) => void, context?: any) => MultiThreadedWallet)' has signatures, but none of those signatures are compatible with each other.
436       this.wallet.on('channelUpdated', (channelResult) => {
                      ~~

src/channel-manager.ts:436:41 - error TS7006: Parameter 'channelResult' implicitly has an 'any' type.
436       this.wallet.on('channelUpdated', (channelResult) => {
                                            ~~~~~~~~~~~~~

Found 2 errors.
```

--- a/packages/server-wallet/server-wallet.api.md
+++ b/packages/server-wallet/server-wallet.api.md
@@ -231,20 +231,14 @@ export function validateServerWalletConfig(config: Record<string, any>): {
 };
 
 // Warning: (ae-forgotten-export) The symbol "SingleThreadedWallet" needs to be exported by the entry point index.d.ts
-// Warning: (ae-forgotten-export) The symbol "MultiThreadedWallet" needs to be exported by the entry point index.d.ts
 //
 // @public (undocumented)
-export type Wallet = SingleThreadedWallet | MultiThreadedWallet;
-
-// @public (undocumented)
-export const Wallet: {
-    create(walletConfig: IncomingServerWalletConfig): Wallet;
-};
-
-// Warning: (ae-forgotten-export) The symbol "ChannelUpdatedEvent" needs to be exported by the entry point index.d.ts
-//
-// @public (undocumented)
-export type WalletEvent = ChannelUpdatedEvent;
+export abstract class Wallet extends SingleThreadedWallet {
+    // Warning: (ae-forgotten-export) The symbol "MultiThreadedWallet" needs to be exported by the entry point index.d.ts
+    //
+    // (undocumented)
+    static create(walletConfig: IncomingServerWalletConfig): SingleThreadedWallet | MultiThreadedWallet;
+}
 
 
 // (No @packageDocumentation comment for this package)

--- a/packages/server-wallet/src/objectives/types.ts
+++ b/packages/server-wallet/src/objectives/types.ts
@@ -4,7 +4,7 @@ import {ChannelResult} from '@statechannels/client-api-schema';
 import {Store} from '../wallet/store';
 import {ChainServiceInterface} from '../chain-service';
 import {Outgoing} from '../protocols/actions';
-import {WalletEvent} from '../wallet';
+import {WalletEvent} from '../wallet/types';
 
 export interface ObjectiveManagerParams {
   store: Store;

--- a/packages/server-wallet/src/wallet/__test__/integration/update-funding.test.ts
+++ b/packages/server-wallet/src/wallet/__test__/integration/update-funding.test.ts
@@ -155,3 +155,46 @@ it('sends the post fund setup when the funding event is provided', async () => {
     channelResults: [{channelId: c.channelId, turnNum: 2}],
   });
 });
+
+it('emits new channel result when the funding event is provided via holdingUpdated', async () => {
+  const c = channel({
+    vars: [
+      stateWithHashSignedBy([alice()])({turnNum: 0}),
+      stateWithHashSignedBy([bob()])({turnNum: 1}),
+    ],
+  });
+  await Channel.query(w.knex).insert(c);
+  const {channelId} = c;
+
+  await ObjectiveModel.insert(
+    {
+      type: 'OpenChannel',
+      participants: c.participants,
+      data: {
+        targetChannelId: c.channelId,
+        fundingStrategy: 'Direct',
+        role: 'app',
+      },
+      status: 'approved',
+    },
+    w.knex
+  );
+
+  const channelUpdatedPromise = new Promise(resolve =>
+    w.on('channelUpdated', ({channelResult}) =>
+      channelResult.channelId === channelId ? resolve(channelResult) : undefined
+    )
+  );
+
+  await w.holdingUpdated({
+    channelId: c.channelId,
+    assetHolderAddress: makeAddress(constants.AddressZero),
+    amount: BN.from(4),
+  });
+
+  const channelResult = await channelUpdatedPromise;
+
+  await expect(Funding.getFundingAmount(w.knex, channelId, AddressZero)).resolves.toEqual('0x04');
+
+  expect(channelResult).toMatchObject({channelId: c.channelId, turnNum: 2});
+});

--- a/packages/server-wallet/src/wallet/index.ts
+++ b/packages/server-wallet/src/wallet/index.ts
@@ -1,20 +1,19 @@
-export {WalletEvent, SingleChannelOutput, MultipleChannelOutput} from './types';
-
 import {IncomingServerWalletConfig} from '../config';
 
+export {SingleChannelOutput, MultipleChannelOutput} from './types';
 import {MultiThreadedWallet} from './multi-threaded-wallet';
 import {SingleThreadedWallet} from './wallet';
 
-export type Wallet = SingleThreadedWallet | MultiThreadedWallet;
-
-export const Wallet = {
-  create(walletConfig: IncomingServerWalletConfig): Wallet {
+export abstract class Wallet extends SingleThreadedWallet {
+  static create(
+    walletConfig: IncomingServerWalletConfig
+  ): SingleThreadedWallet | MultiThreadedWallet {
     if (walletConfig?.workerThreadAmount && walletConfig.workerThreadAmount > 0) {
       return MultiThreadedWallet.create(walletConfig);
     } else {
       return SingleThreadedWallet.create(walletConfig);
     }
-  },
-};
+  }
+}
 
 export * from '../config';

